beamer/config/state.py

```diff
@@ -83,6 +76,65 @@ def _check_tokens(self) -> Generator:
                 loc = get_alias(self).token_addresses
                 yield loc, f"missing address for token {name}"
 
+    @staticmethod
+    def _convert_chain_ids(data: dict) -> None:
+        # convert chain ID strings to integers, since the schema expects
+        # integers
+        if "RequestManager" in data:
+            chains = data["RequestManager"].get("chains")
+            if chains is not None:
+                new_chains = {}
+                for chain_id, value in chains.items():
+                    try:
+                        chain_id = ChainId(int(chain_id))
+                    except ValueError as exc:
+                        raise ValidationError(f"invalid chain ID: {chain_id}") from exc
+                    new_chains[chain_id] = value
+                data["RequestManager"]["chains"] = new_chains
+
+    def clone(self: T) -> T:
+        return copy.deepcopy(self)
+
+
+@dataclass
+class DesiredConfiguration(_BaseConfiguration):
+    @staticmethod
+    def from_file(path: Path) -> "DesiredConfiguration":
+        with open(path, "rt") as f:
+            data = json.load(f)
+
+        _BaseConfiguration._convert_chain_ids(data)
+        try:
+            config = apischema.deserialize(DesiredConfiguration, data)
+        except apischema.ValidationError as exc:
+            raise ValidationError from exc
+        return config
+
+    def to_file(self, path: Path) -> None:
+        data = apischema.serialize(self)
+        with open(path, "wt") as f:
+            json.dump(data, f, indent=4)
+
+    def to_config(self, block: BlockNumber) -> "Configuration":
+        data = apischema.serialize(self, no_copy=False)
+        data["block"] = block
+        return apischema.deserialize(Configuration, data)
```

Comment:
not sure if we need try/except here like other places

Reply:
We don't because this is already a live instance which was validated once upon loading. (It could happend that it is modified in an invalid way, but then we will get an exception, which is what we want anyway.)